### PR TITLE
Add Custom support for ViewPager

### PR DESCRIPTION
**Pull request by niqo01**
Hello,

I've added two methods to customize the touch behavior:
- setDragBezelFullScreen (Set the touch area to full width)
- setDragBezelEnabled (Enable/disable dragging)

This is useful when using with a ViewPager in the content view and allow to reproduce the Verge app behavior.

How to:
When using the menu with a ViewPager, set setDragBezelFullScreen to true
If the View pager current position is the first page call setDragBezelEnabled to true otherwise set setDragBezelEnabled to false.

Thank you for this awesome library!

Nicolas

### CODE REVIEW

**Comment by SimonVT** [comment]
Thanks for the contribution.
It'd be nice if you could add a sample to the samples project. There are also several formatting errors in the commit that should be fixed before this is merged.

**Comment by niqo01** [comment]
Hello,

I am not use to contribute so thank you for your answer and guidelines.
Your reviews are indeed exactly what I should have done.
I will try to fix this commit asap,  however I am pretty overbooked right
now.
Mainly because I am "the" (only one) Android developer at CouchSurfing and
we need and are looking for 3 other Android guys at least.
Just in case you would be interested ^^

I should be able to do another push within the next 2 or 3 weeks.

Sincerely,

Nicolas

On Thu, Oct 11, 2012 at 9:03 AM, Simon Vig Therkildsen <
notifications@github.com> wrote:

> Thanks for the contribution.
> It'd be nice if you could add a sample to the samples project. There are
> also several formatting errors in the commit that should be fixed before
> this is merged.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/SimonVT/android-menudrawer/pull/20#issuecomment-9346183.

## 

## Nicolas Milliard

Linux Foundation Member

## http://www.linuxfoundation.org

**Comment by SimonVT** [comment]
That's fine, no rush :) Also, move the public methods up with the other public methods.

**Comment by niqo01** [comment]
I've updated the branch with your recommendations. I was wondering if you can review those changes.

**Comment by SimonVT** [comment]
Could you squash it into a single commit? Thanks

**Comment by niqo01** [comment]
I usually use eclipse format command with default settings. However your project doesn't follow those defaults (Ex: spaces instead of tabs, ...). I manually fix the formatting errors: "){" to ") {". Is there other formatting issues I am not following?

**Comment by JakeWharton** [comment]
You should probably adhere to the AOSP coding style [defined here](http://s.android.com/source/code-style.html) rather than whatever arbitrary defaults some IDE uses.

**Comment by niqo01** [comment]
I am using a custom coding style for my company but will seriously consider switching to the AOSP one. Thanks @JakeWharton for pointing this useful information. (Also the necessary eclipse configuration files are provided by AOSP: http://source.android.com/source/using-eclipse.html, Awesome). 

I have checked my last commit with this format and the code is clean. 
However and for information, using the AOSP eclipse format, several issues were detected outside the scope of my commit concerning the missing @Override on some methods and imports order.

**Comment by SimonVT** [comment]
There are empty lines with space indentation, several methods in MenuDrawer.java still uses tab indentation, the entire sample uses tab indentation. The diff on github makes this quite easy to see.

The import order defined in aosp makes sense in aosp, the one I use makes sense in my libraries (for the same reasons they listed). Anyway, the ordering should always follow the rest of the library.

I'll try it out this weekend.

**Comment by niqo01** [comment]
@SimonVT  I agree with you. I should have take more care about this library coding style from the beginning. Sorry for those back and force. I've replaced all tabs by spaces and changed the if condition.

File: samples/src/net/simonvt/menudrawer/samples/ViewPagerSample.java [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+    private static class Category {
+
+        String mTitle;
+
+        Category(String title) {
+            mTitle = title;
+        }
+    }
+
+    private class MenuAdapter extends BaseAdapter {
```
Comment (JakeWharton) [inline samples/src/net/simonvt/menudrawer/samples/ViewPagerSample.java:173]: Extending `ArrayAdapter` will save you from implementing a few of these methods.

File: samples/src/net/simonvt/menudrawer/samples/ViewPagerSample.java [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+
+        Item(String title, int iconRes) {
+            mTitle = title;
+            mIconRes = iconRes;
+        }
+    }
+
+    private static class Category {
+
+        String mTitle;
```
Comment (JakeWharton) [inline samples/src/net/simonvt/menudrawer/samples/ViewPagerSample.java]: Should be `final`.

File: samples/src/net/simonvt/menudrawer/samples/ViewPagerSample.java [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+            return;
+        }
+
+        super.onBackPressed();
+    }
+
+    private static class Item {
+
+        String mTitle;
+        int mIconRes;
```
Comment (JakeWharton) [inline samples/src/net/simonvt/menudrawer/samples/ViewPagerSample.java]: Should both be `final`.

File: library/src/net/simonvt/widget/MenuDrawer.java [hunk 4]
```diff
@@ -699,6 +726,25 @@ public void setDragMode(int dragMode) {
...
 
+    /**
+     * Returns the touch mode.
+     */
+    public int getTouchMode() {
+        return mTouchMode;
+    }
+
+    /**
+     * Sets the drawer touch mode. Possible values are {@link #TOUCH_MODE_NONE}, {@link #TOUCH_MODE_BEZEL} or {@link #TOUCH_MODE_FULLSCREEN}
```
Comment (JakeWharton) [inline library/src/net/simonvt/widget/MenuDrawer.java]: Line too long (120 char max)

File: samples/src/net/simonvt/menudrawer/samples/SamplesActivity.java [hunk 5]
```diff
@@ -28,6 +28,7 @@ public void onCreate(Bundle savedInstanceState) {
         mAdapter.addSample("ActionBar overlay sample", "A window sample, where the ActionBar is an overlay",
                 ActionBarOverlaySample.class);
         mAdapter.addSample("Right menu", "The menu is positioned to the right of the content", RightMenuSample.class);
+        mAdapter.addSample("Touch Mode", "The menu touch behavior change according to different content view state (Ex: View Pager)", ViewPagerSample.class);
```
Comment (JakeWharton) [inline samples/src/net/simonvt/menudrawer/samples/SamplesActivity.java]: Line too long.

**Comment by niqo01** [comment]
@JakeWharton thanks for your review. I've pushed a few changes according to your review except for the "Extending ArrayAdapter" one. Because this piece of code was imported from an other sample class ContentSample.java. Even though this is a minor change, I prefer to let @SimonVT decided to change this in its samples or not.

**Comment by SimonVT** [comment]
The code looks good, but there's still some checkstyle violations (a checkstyle.xml is now included in the master branch).

MenuDrawer.java : 11 item(s)
First sentence should end with a period. (126:0)
Line has trailing spaces. (130:0)
First sentence should end with a period. (131:0)
Line has trailing spaces. (135:0)
First sentence should end with a period. (136:0)
First sentence should end with a period. (272:0)
Line has trailing spaces. (412:0)
Line has trailing spaces. (737:0)
First sentence should end with a period. (863:0)
Line has trailing spaces. (870:0)
')' is preceded with whitespace. (1,126:56)

SamplesActivity.java : 2 item(s)
'+' should be on a new line. (31:98)
File contains tab characters (this is the first instance). (32:9)

ViewPagerSample.java : 2 item(s)
Name '_class' must match pattern '^[a-z][a-zA-Z0-9]_$'. (270:30)
Name '_args' must match pattern '^[a-z][a-zA-Z0-9]_$'. (270:45)

Also, the support library used should be added to samples/libs/.

File: library/src/net/simonvt/widget/MenuDrawer.java [hunk 6]
```diff
@@ -248,12 +263,17 @@
...
     /**
      * The touch area width of the drawer in px.
      */
-    protected int mDragBezelSize;
+    protected int mTouchWidth;
+
+    /**
+     * Save the measured width to re compute touch mode changes
+     */
+    protected int mMeasuredWidth;
```
Comment (JakeWharton) [inline library/src/net/simonvt/widget/MenuDrawer.java]: This is already cached [in the `View` class](https://github.com/android/platform_frameworks_base/blob/master/core/java/android/view/View.java#L8589-8591). Just call `getMeasuredWidth()` when you need it.

**Comment by SimonVT** [comment]
More comments [here](https://github.com/niqo01/android-menudrawer/commit/1bb19c58e4a8e919bc1083b23603d2f10184442a).

**Comment by niqo01** [comment]
@JakeWharton  @SimonVT . I just pushed changes according to your reviews. The only review left is the "extends ArrayAdapter " but let @SimonVT decided to do the change on the samples later or not.

**Comment by SimonVT** [comment]
Thanks for being patient, and quick with the fixes :)

**Comment by SimonVT** [comment]
Thanks! :)

**Comment by niqo01** [comment]
Thank you for being patient with me too and for this awesome lib.

**Pull request merged**